tests/foreman/api/test_contentviewversion.py

```diff
@@ -42,52 +41,42 @@ def test_delete_version(self):
 
         """
         key_content = read_data_file(ZOO_CUSTOM_GPG_KEY)
-        org_id = entities.Organization().create_json()['id']
-        gpgkey_id = entities.GPGKey(
+        org = entities.Organization().create()
+        gpgkey = entities.GPGKey(
             content=key_content,
-            organization=org_id
-        ).create_json()['id']
+            organization=org,
+        ).create()
         # Creates new product without selecting GPGkey
-        product_id = entities.Product(
-            organization=org_id
-        ).create_json()['id']
+        product = entities.Product(organization=org).create()
         # Creates new repository with GPGKey
         repo = entities.Repository(
+            gpg_key=gpgkey,
+            product=product,
             url=FAKE_1_YUM_REPO,
-            product=product_id,
-            gpg_key=gpgkey_id,
         ).create()
         # sync repository
         repo.sync()
         # Create content view
-        content_view = entities.ContentView(
-            organization=org_id
-        ).create()
+        content_view = entities.ContentView(organization=org).create()
         # Associate repository to new content view
-        client.put(
-            content_view.path(),
-            {u'repository_ids': [repo.id]},
-            auth=get_server_credentials(),
-            verify=False,
-        ).raise_for_status()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
```

Comment:
We can use either, but I prefer the `update` method over the `set_repository_ids` method. The `set_repository_ids` method was added before the `update()` method became available, and that's the only reason it exists.